### ISSUE TITLE
ENH: GeoDataArray does not support setting no_data

### ISSUE THREAD

**Issue by Tjalling-dejong** [Deltares/hydromt#431]
### Kind of request

Adding new functionality

### Enhancement Description

In HydroMT's current state vector GeoBase class does not support setting no_data values. It is possible to set the no data values for XRasterBase.  It should also be possible for GeoBase class. 

The issue arises when a geodataset has a specified nodata value in the datacatalog. The nodata is not set when using get_data. 

### Use case

The issue arises when a geodataset has a specified nodata value in the datacatalog. The nodata is not set when using get_geodataset.  

For instance this test cannot assert the nodata value of the geodataset:
 ```
def test_set_nodata_geodataset(artifact_data: DataCatalog):
    gtsm_dict = {"gtsmv3_eu_era5": artifact_data.sources["gtsmv3_eu_era5"].to_dict()}
    gtsm_dict["gtsmv3_eu_era5"].update(dict(nodata=-99))
    datacatalog = DataCatalog()
    datacatalog.from_dict(gtsm_dict)
    datacatalog.get_geodataset("gtsmv3_eu_era5")
    # assert gtsm_geodataarray.vector.nodata == -99 TODO assertion cannot be made
    # since set_noda for GeoDataArray.vector does not exist
```

**Comment by DirkEilander** [comment]
Currently it is possible to set nodata values for `RasterDataset` objects, not the base objects that you refer to.
Similarly, it would be a good enhencement to add a `nodata` property and `set_nodata` method to the `GeoDataArray` class.

**Comment by savente93** [comment]
This one seems to have been resolved in #334
